scheduler/scheduler_test.go

```diff
@@ -128,48 +131,194 @@ func TestScheduler(t *testing.T) {
-						}
-						if cfg != stack.Config() {
-							t.Errorf("Failed to get stack with configuration, want: %v, got: %v", cfg, stack)
+
+						cfg := lf.Config()
+						queue := lf.GetQueue()
+						if queue == nil {
+							t.Errorf("Queue is nil")
 						}
 
-						if lf.Key() != key {
-							t.Errorf("Failed to get the right key: %s, expected: %s", lf.Key(), key)
+						if cfg != queue.Config() {
+							t.Errorf("Failed to get queue with configuration, want: %v, got: %v", cfg, queue)
 						}
-						k := lf.Key()
-						stacksMap[k] = append(stacksMap[k], stack)
+
+						queuesMap[key] = append(queuesMap[key], queue)
 					}
 				}
-				if len(stacksMap[key]) != len(group) {
-					t.Errorf("Failed to get the right group size %v != %v", len(stacksMap[key]), len(group))
+
+				if len(queuesMap[key]) != len(group) {
+					t.Errorf("Failed to get the right group size %v != %v", len(queuesMap[key]), len(group))
 				}
 			}
-			// check pointers to stack are the same for same group
-			for k, stacks := range stacksMap {
-				firstStack := stacks[0]
-				for _, stack := range stacks {
-					if stack != firstStack {
-						t.Errorf("Unexpected different stack in group: %s", k)
+			// check pointers to queue are the same for same group
+			for k, stacks := range queuesMap {
+				firstQueue := stacks[0]
+				for _, queue := range stacks {
+					if queue != firstQueue {
+						t.Errorf("Unexpected different queue in group: %s", k)
 					}
 				}
 			}
-			// check pointers to stack of different groups are different
-			diffStacks := make(map[*scheduler.Stack]struct{})
-			for _, stacks := range stacksMap {
-				diffStacks[stacks[0]] = struct{}{}
+			// check pointers to queue of different groups are different
+			diffQueues := make(map[*scheduler.Queue]struct{})
+			for _, stacks := range queuesMap {
+				diffQueues[stacks[0]] = struct{}{}
 			}
-			if len(diffStacks) != len(stacksMap) {
-				t.Error("Unexpected got pointer to the same stack for different group")
+			if len(diffQueues) != len(queuesMap) {
+				t.Error("Unexpected got pointer to the same queue for different group")
 			}
 		})
 	}
 
 }
+
+func TestConfig(t *testing.T) {
+	waitForStatus := func(t *testing.T, q *scheduler.Queue, s scheduler.QueueStatus) {
+		timeout := time.After(120 * time.Millisecond)
+		for {
+			if q.Status() == s {
+				return
+			}
+
+			select {
+			case <-timeout:
+				t.Fatal("failed to reach status")
+			default:
+			}
+		}
+	}
+
+	initTest := func(doc string) (*scheduler.Registry, *routing.Routing, *testdataclient.Client) {
+		cli, err := testdataclient.NewDoc(doc)
+		if err != nil {
+			t.Fatalf("Failed to create a test dataclient: %v", err)
+		}
+
+		reg := scheduler.NewRegistry()
+		ro := routing.Options{
+			SignalFirstLoad: true,
+			FilterRegistry:  builtin.MakeRegistry(),
+			DataClients:     []routing.DataClient{cli},
+			PostProcessors: []routing.PostProcessor{
+				reg,
+			},
+		}
+
+		rt := routing.New(ro)
+		<-rt.FirstLoad()
+		return reg, rt, cli
```

Comment:
could also return a func close, that close reg and rt, then you don't have a double defer